renderer/HardwareBaseGPU.cpp

```diff
@@ -434,3 +438,319 @@ void rend_GetStatistics(tRendererStats *stats) {
+
+    vData->uv.s = pnt->p3_u;
+    vData->uv.t = pnt->p3_v;
+    vData->uv.r = 0.0f;
+    vData->uv.w = 1.0f;
+
+    // Finally, specify a vertex
+    vData->pos = pnt->p3_vecPreRot;
+  }
+
+  // And draw!
+  gpu_RenderPolygon(&vArray[0], nv);
+
+  // If there is a lightmap to draw, draw it as well
+  if (gpu_Overlay_type != OT_NONE) {
+    return; // Temp fix until I figure out whats going on
+    Int3(); // Shouldn't reach here
+  }
+}
+
+#endif // DEDICATED_ONLY
+
+// Takes nv vertices and draws the polygon defined by those vertices.  Uses bitmap "handle"
+// as a texture
+void rend_DrawMultitexturePolygon3D(int handle, g3Point **p, int nv, int map_type) {
+  g3Point *pnt;
+  int i, fr, fg, fb;
+  float alpha;
+
+  float one_over_square_res = 1.0 / GameLightmaps[gpu_Overlay_map].square_res;
+  float xscalar = (float)GameLightmaps[gpu_Overlay_map].width * one_over_square_res;
+  float yscalar = (float)GameLightmaps[gpu_Overlay_map].height * one_over_square_res;
+
+  ASSERT(nv < 100);
+
+  if (gpu_state.cur_light_state == LS_NONE) {
+    fr = GR_COLOR_RED(gpu_state.cur_color);
+    fg = GR_COLOR_GREEN(gpu_state.cur_color);
+    fb = GR_COLOR_BLUE(gpu_state.cur_color);
+  }
+
+  alpha = gpu_Alpha_multiplier * gpu_Alpha_factor;
+
+  PosColorUV2Vertex *vData = &vArray2[0];
+
+  // Specify our coordinates
+  for (i = 0; i < nv; i++, vData++) {
+    pnt = p[i];
+    ASSERT(pnt->p3_flags & PF_ORIGPOINT);
+
+    if (gpu_state.cur_alpha_type & ATF_VERTEX)
+      alpha = pnt->p3_a * gpu_Alpha_multiplier * gpu_Alpha_factor;
+
+    // If we have a lighting model, apply the correct lighting!
+    if (gpu_state.cur_light_state != LS_NONE) {
+      // Do lighting based on intesity (MONO) or colored (RGB)
+      if (gpu_state.cur_color_model == CM_MONO) {
+        vData->color.r = pnt->p3_l;
+        vData->color.g = pnt->p3_l;
+        vData->color.b = pnt->p3_l;
+        vData->color.a = alpha;
+      } else {
+        vData->color.r = pnt->p3_r;
+        vData->color.g = pnt->p3_g;
+        vData->color.b = pnt->p3_b;
+        vData->color.a = alpha;
+      }
+    } else {
+      vData->color.r = 1;
+      vData->color.g = 1;
+      vData->color.b = 1;
+      vData->color.a = alpha;
+    }
+
+    /*
+    // Texture this polygon!
+    float texw=1.0/(pnt->p3_z+Z_bias);
+    vData->uv0.s=pnt->p3_u*texw;
+    vData->uv0.t=pnt->p3_v*texw;
+    vData->uv0.r=0;
+    vData->uv0.w=texw;
+
+    vData->uv1.s=pnt->p3_u2*xscalar*texw;
+    vData->uv1.t=pnt->p3_v2*yscalar*texw;
+    vData->uv1.r=0;
+    vData->uv1.w=texw;
+    */
+    vData->uv0.s = pnt->p3_u;
+    vData->uv0.t = pnt->p3_v;
+    vData->uv0.r = 0.0f;
+    vData->uv0.w = 1.0f;
+
+    vData->uv1.s = pnt->p3_u2 * xscalar;
+    vData->uv1.t = pnt->p3_v2 * yscalar;
+    vData->uv1.r = 0.0f;
+    vData->uv1.w = 1.0f;
+
+    // Finally, specify a vertex
+    /*
+    vData->pos.x=pnt->p3_sx+x_add;
+    vData->pos.y=pnt->p3_sy+y_add;
+    vData->pos.z = -std::max(0,std::min(1.0,1.0-(1.0/(pnt->p3_z+Z_bias))));
```

Comment:
not in the original function?

Reply:
Yeah, this is from the original, I didn't add any commented code on purpose AFAIK.